lib/instrumentation/transaction.js

```diff
@@ -12,8 +12,11 @@ var Timer = require('./timer')
 
 module.exports = Transaction
 
-function Transaction (agent, name, type, traceparent) {
-  this.context = TraceContext.startOrResume(traceparent, agent._conf)
+function Transaction (agent, name, type, traceContext) {
+  this.context = traceContext instanceof TraceContext
+    ? traceContext.child()
+    : TraceContext.startOrResume(traceContext, agent._conf) // string | null expected
```

Comment:
Maybe just make `TraceContext.startOrResume(...)` accept another `TraceContext` instance to call `traceContext.child()` on? 🤔

Reply:
Created separate PR: #826